bundles/org.openhab.binding.modbus.kermi/src/main/java/org/openhab/binding/modbus/kermi/internal/handler/KermiXcenterThingHandler.java

```diff
@@ -0,0 +1,666 @@
+                        this::handleStateResult, this::handleStateFailure);
+
+                pollTasks.add(statePoller);
+
+                // default polling speed
+                ModbusReadRequestBlueprint chargingcircuitRequest = new ModbusReadRequestBlueprint(slaveId,
+                        ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, CHARGING_CIRCUIT_REG_START,
+                        CHARGING_CIRCUIT_REG_SIZE, 3);
+                chargingCircuitPoller = localComms.registerRegularPoll(chargingcircuitRequest, localConfig.refresh, 0,
+                        this::handleChargingCircuitResult, this::handleChargingCircuitFailure);
+
+                pollTasks.add(chargingCircuitPoller);
+
+                ModbusReadRequestBlueprint energySourceRequest = new ModbusReadRequestBlueprint(slaveId,
+                        ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, ENERGY_SOURCE_REG_START, ENERGY_SOURCE_REG_SIZE,
+                        3);
+                energySourcePoller = localComms.registerRegularPoll(energySourceRequest, localConfig.refresh, 0,
+                        this::handleEnergySourceResult, this::handleEnergySourceFailure);
+
+                pollTasks.add(energySourcePoller);
+
+                ModbusReadRequestBlueprint powerRequest = new ModbusReadRequestBlueprint(slaveId,
+                        ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, POWER_REG_START, POWER_REG_SIZE, 3);
+                powerPoller = localComms.registerRegularPoll(powerRequest, localConfig.refresh, 0,
+                        this::handlePowerResult, this::handlePowerFailure);
+
+                pollTasks.add(powerPoller);
+
+                if (localConfig.pvEnabled) {
+                    ModbusReadRequestBlueprint pvRequest = new ModbusReadRequestBlueprint(slaveId,
+                            ModbusReadFunctionCode.READ_MULTIPLE_REGISTERS, PV_MODULATION_REG_START,
+                            PV_MODULATION_REG_SIZE, 3);
+                    pvPoller = localComms.registerRegularPoll(pvRequest, localConfig.refresh, 0, this::handlePvResult,
+                            this::handlePvFailure);
+
+                    pollTasks.add(pvPoller);
+                }
+
+            } // else state handling performed in connectEndPoint function
+        });
+    }
+
+    /**
+     * Get a reference to the modbus endpoint
+     */
+    private @Nullable ModbusCommunicationInterface connectEndpoint() {
+        if (comms != null) {
+            return comms;
+        }
+
+        ModbusEndpointThingHandler slaveEndpointThingHandler = getEndpointThingHandler();
+        if (slaveEndpointThingHandler == null) {
+            @SuppressWarnings("null")
+            String label = Optional.ofNullable(getBridge()).map(b -> b.getLabel()).orElse("<null>");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                    String.format("Bridge '%s' is offline", label));
```

Comment:
The bridge label is not that important is it? I would opt to just set the state. 
```suggestion
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
```